ksql-engine/src/main/java/io/confluent/ksql/materialization/Materialization.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.materialization;
+
+import io.confluent.ksql.model.WindowType;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import java.util.Optional;
+
+public interface Materialization {
+
+  /**
+   * @return The schema of the materialization
+   */
+  LogicalSchema schema();
+
+  /**
+   * @return service for locating which node holds specific data.
+   */
+  Locator locator();
```

Comment:
Yep, I know what you mean.  This was just how the code fell out during the prototyping, and given the tight deadlines I'm focusing more on functionality than refactoring. Totally open for changing this once the core functionality (and tight deadline) is met.  

I've been thinking that it might be better structured as:

- Engine stores mapping on materialized tables to a `Locator`.
- You can ask the locator for who owns a key, (or in the future keys, all)
- Locator will response with abstraction that either routes to remote node, or locally handles request.

This was my plan, and I think fits better with your thinking too - but would love to hear your thoughts.